swingset/src/main/java/com/nqadmin/swingset/SSDBComboBox.java

```diff
@@ -1504,31 +1478,52 @@ public void setSelectedItem(final Object _value) {
 	 *
 	 * @param _value value to set as currently selected.
 	 */
+	//
+	// TODO: WHAT IS THIS SUPPOSED TO DO IF BOTH O,O2 are set?
+	//
+	// TODO: thought this was deprecated/removed. Maybe that was SSComboBox
+	// 
 	public void setSelectedStringValue(final String _value) {
```

Comment:
There's some considerations.

Reply:
I don't even see it in SSComboBox 3 or 4. It was in SSDBComboBox 3.x. Will add it to my list of things to check.

Reply:
In 3, this method was one line of code that set the text in the underlying bound JTextField:
this.textField.setText(_value);

I feel like something has gotten out of hand one this one.

Considering the old days of primary keys (Mappings) and display strings (Options) and you (the developer) want to set the combo based on a target display string (Option), how would you do that? I think we've discussed that it's a dangerous methods because there could be more than one string match (Option) so it may not behave as intended.

I would find/set the first Option that matches or log a warning and setSelectedIndex(-1) or setSelectedItem(null).

If this is not used a lot then maybe we can deprecate.

Reply:
I'm thinking this is the right thing to do, but that depends on the callers intent. It's all about what to do if the item isn't found. Should it just be put into the editable combo box?
```
					Object item;
					if (index != -1) {
						item = remodel.get(index);
					} else {
						//item = nullItem;
						// Didn't find it in the list,
						// so just use it for setSelectedItem.
						item = _value;
						logger.warn(() -> String.format("%s: Could not find a corresponding combobox item for %s. Using it for selectedItem.", getColumnForLog(), _value));
					}
					
					// setSelectedIndex()->setSelectedItem() For glazed reliability.
					// setSelectedIndex(index);

					setSelectedItem(item);
```
So either a listItem that matches something in the itemList is selected or whatever string was passed in is put into the editable combo box according to setSelectedItem rules.

Reply:
I'm still confused. And what if both O and O2 are set?

Reply:
We use this 4 times each with a string so presumably not the multi column scenario. If O2 is set then I would throw an exception.

I think this method could be helpful if you have a relatively small list and no O2. Probably add a bit to JavaDoc to indicate it will fail if you configure O2 and could give unexpected results if there are multiple matches. Maybe move it to SSBaseComboBox? It seems it would be more useful for an SSComboBox.

Could deprecate in favor of a setSelectedOption()?

Reply:
I don't think putting it in the combo editor helps, although maybe that give the user some useful feedback?

Reply:
At this point, unless there's a clear reason, I'm thinking that we give things the feel of a combo box. So if the app does `setSelectedOption("bogus_value")`, and "bogus_value" is not a known option, then do setSelectedItem("bogus_value"). It's on the app.

Reply:
> then do setSelectedItem("bogus_value"). It's on the app.

When stuff migrates into SSBaseCombo, then it becomes more clear what changes/adjustments are to expected behavior. Then it might make sense to specialize things more.

Reply:
I'm good with consistency and I'll squawk if it breaks things.